bson/primitive/decimal.go

```diff
@@ -52,13 +59,13 @@ func (d Decimal128) String() string {
 	if d.h>>61&3 == 3 {
 		// Bits: 1*sign 2*ignored 14*exponent 111*significand.
 		// Implicit 0b100 prefix in significand.
-		e = int(d.h>>47&(1<<14-1)) - 6176
+		e = int(d.h>>47&(1<<14-1)) + MinDecimal128Exp
 		//h = 4<<47 | d.h&(1<<47-1)
 		// Spec says all of these values are out of range.
 		h, l = 0, 0
 	} else {
 		// Bits: 1*sign 14*exponent 113*significand
-		e = int(d.h>>49&(1<<14-1)) - 6176
+		e = int(d.h>>49&(1<<14-1)) + MinDecimal128Exp
```

Comment:
I'd rename `e` to `exp` to stay consistent with the implementation for `BigInt()`. This should also let you remove the comment above explaining that `e` is an exponent.